internal/cache/util/util_test.go

```diff
@@ -233,3 +234,37 @@ func (ut *utilTest) Test_IsCacheHandleValid_False() {
 		ExpectFalse(IsCacheHandleInvalid(errors.New(errMsg)))
 	}
 }
+
+func TestCheckPermissionsForShouldNotReturnAnyErrorWhenDirectoryExists(t *testing.T) {
+	base := "./" + string(testutil.GenerateRandomBytes(4))
+	dirPath := base + "/" + "path/cachedir"
+	dirCreationErr := os.MkdirAll(dirPath, 0700)
+	defer os.RemoveAll(base)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	ExpectEq(nil, dirCreationErr)
```

Comment:
we should use AssertEq instead of ExpectEq. The difference is that AssertEq doesn't execute the unit test further if False. so e.g. in this case it is good to just stop this unit test if ExpectEq(nil, dirCreationErr) is not True.

Also, we should move ExpectEq(nil, dirCreationErr) before err := CreateCacheDirectoryIfNotPresentAt(dirPath) because ExpectEq(nil, dirCreationErr) has to be true for testing err := CreateCacheDirectoryIfNotPresentAt(dirPath)

Reply:
updated as per comments, was trying previously to make tests formatted as per [ref](https://docs.telerik.com/devtools/justmock/basic-usage/arrange-act-assert#:~:text=Arrange%2FAct%2FAssert%20(AAA,more%20natural%20and%20convenient%20way.)